pkg/virt-handler/cgroup/cgroup_v1_manager.go

```diff
@@ -109,3 +114,94 @@ func getCurrentlyDefinedRules(runcManager runc_cgroups.Manager) ([]*devices.Rule
 func (v *v1Manager) GetCpuSet() (string, error) {
 	return getCpuSetPath(v, "cpuset.cpus")
 }
+
+func rw_filecontents(fReadPath string, fWritePath string) (err error) {
+	rFile, err := os.Open(fReadPath)
+	if err != nil {
+		return fmt.Errorf("Open failed: %s (%v)", fReadPath, err)
+	}
+	defer rFile.Close()
+
+	wFile, err := os.OpenFile(fWritePath, os.O_RDWR, 0755)
+	if err != nil {
+		return fmt.Errorf("OpenFile failed: %s (%v)", fWritePath, err)
+	}
+	defer wFile.Close()
+
+	data := make([]byte, 20000)
+	count, err := rFile.Read(data)
+	if err != nil {
+		return fmt.Errorf("Read failed: %s (%v)", fWritePath, err)
+	}
+
+	_, err = wFile.Write(data[:count])
+	if err != nil {
+		return fmt.Errorf("Write failed: %s (%v)", fWritePath, err)
+	}
+
+	return nil
+}
+
+// Attach TID to cgroup. Optionally on a subcgroup of
+// the pods control group (if subcgroup != nil).
+func (v *v1Manager) AttachTID(subSystem string, subCgroup string, tid int) error {
+	cgroupPath, err := v.GetBasePathToHostSubsystem(subSystem)
+	if err != nil {
+		return err
+	}
+	if subCgroup != "" {
+		cgroupPath = filepath.Join(cgroupPath, subCgroup)
+	}
```

Comment:
In case the subCgroup does not exist, the error from runc_cgroups.WriteFile will be returned:


// WriteFile writes data to a cgroup file in dir.
// It is supposed to be used for cgroup files only.
func WriteFile(dir, file, data string) error {
        fd, err := OpenFile(dir, file, unix.O_WRONLY)
        if err != nil {
                return err
        }

Whose error format is:

open /path/to/file: error string

Reply:
We don't. Again: In case the subCgroup does not exist, the error from runc_cgroups.WriteFile will be returned:


// WriteFile writes data to a cgroup file in dir.
// It is supposed to be used for cgroup files only.
func WriteFile(dir, file, data string) error {
        fd, err := OpenFile(dir, file, unix.O_WRONLY)
        if err != nil {
                return err
        }

Whose error format is:

open /path/to/file: error string

Reply:
See comment here: https://github.com/kubevirt/kubevirt/pull/8233#discussion_r1015106842